### PR TITLE
[NVQC] Add logging info for timeout 

### PR DESCRIPTION
**Pull request by 1tnguyen**
### Description

- Retrieve timeout info from NVQC deployment (container environment variables).

- Log timeout limit alongside the NVQC info line.

- Warn users when timeout is approaching (less than 5 min remained).

### CODE REVIEW

File: python/runtime/utils/PyRestRemoteClient.cpp [hunk 1]
```diff
@@ -371,15 +383,29 @@ class PyNvcfRuntimeClient : public PyRestRemoteClient {
...
+            std::chrono::duration_cast<std::chrono::seconds>(
+                std::chrono::system_clock::now() - requestStartTime)
+                .count();
         if (m_logLevel > LogLevel::None)
           cudaq::log("Polling NVQC result data for Request Id {}", reqId);
+        // Warns if the remaining time is less than this threshold.
+        constexpr int TIMEOUT_WARNING_SECS = 5 * 60; // 5 minutes.
+        const int remainingSecs = std::max(
+            m_availableFuncs[m_functionId].timeoutSecs - elapsedTimeSecs, 0);
+        if (remainingSecs < TIMEOUT_WARNING_SECS)
```
Comment (1tnguyen) [inline python/runtime/utils/PyRestRemoteClient.cpp:404]: @bmhowe23  I wonder whether we should log this "Approaching the wall time limit ..." line only once or repeat (like this).
Please let me know if you have any recommendations.

Comment (bmhowe23) [reply python/runtime/utils/PyRestRemoteClient.cpp]: > @bmhowe23 I wonder whether we should log this "Approaching the wall time limit ..." line only once or repeat (like this). Please let me know if you have any recommendations.

I put some suggestions in 69008e3f3.

**Review by bmhowe23 — APPROVED** [review]

File: python/runtime/utils/PyRestRemoteClient.cpp [hunk 2]
```diff
@@ -38,8 +38,15 @@ class PyNvcfRuntimeClient : public PyRestRemoteClient {
   std::string m_functionId;
   // NVCF version Id of that function to use
   std::string m_functionVersionId;
-  // Available functions: function Id to number of GPUs mapping
-  using DeploymentInfo = std::unordered_map<std::string, std::size_t>;
+  // Information about function deployment from environment variable info.
+  struct FunctionEnvironments {
+    // These configs should be positive numbers.
```
Comment (schweitzpgi) [inline python/runtime/utils/PyRestRemoteClient.cpp:43]: Positive or non-negative?

Comment (1tnguyen) [reply python/runtime/utils/PyRestRemoteClient.cpp:43]: They should be positive number.

**Pull request merged**